### ISSUE TITLE
[COGNITION]: Buttons - All buttons SHOULD have a darker shade on hover and focus

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/vets-design-system-documentation#235]
@1Copenut commented on [Wed May 13 2020](https://github.com/department-of-veterans-affairs/va.gov-team/issues/9087)

# [508-defect-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-3)

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️. Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

Our `button[class="va-button-primary"]` becomes a darker green shade on `:hover` and `:focus` pseudo-states. Our regular blue button and secondary outlined button do not become a darker shade on `:focus`. This could be improved for users who rely on the keyboard to navigate or use those color change cues for identifying the element with focus. Screenshots attached below.

## Point of Contact

**VFS Point of Contact:** _Trevor_

## Acceptance Criteria

- [x] Blue button has the same background color on hover and focus
- [x] Blue button retains the yellow halo on focus
- [x] Outlined secondary button has the same darker border and text on hover and focus
- [x] Outlined secondary button retains the yellow halo on focus
- [x] Create and/or modify unit tests for this component to include an aXe scan

## WCAG or Vendor Guidance (optional)
* [Focus Visible: Understanding SC 2.4.7](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-visible.html)

## Screenshots or Trace Logs

![Screen Shot 2020-05-13 at 4 57 11 PM](https://user-images.githubusercontent.com/934879/81871145-d73e9280-953c-11ea-8de6-204432298f1f.png)

---
![Screen Shot 2020-05-13 at 4 57 19 PM](https://user-images.githubusercontent.com/934879/81871155-dd347380-953c-11ea-83aa-adb123e98f57.png)

---
![Screen Shot 2020-05-13 at 4 57 40 PM](https://user-images.githubusercontent.com/934879/81871170-e3c2eb00-953c-11ea-94f0-9453723ad26e.png)

**Comment by rjohnson2011** [comment]
@GnatalieH Looking to get your input on the design aspect of this ticket. Do you have any guidance on how to differentiate between focus and hover states within best practices of VA UI/UX Design?

**Comment by GnatalieH** [comment]
@rjohnson2011 I will have a look at this and try and give you an answer in the next couple of days. There is an issue that exists to address that our focus halo isn't accessible against a light background, but that's sort of adjacent to this ticket's needs. [Here is a link](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/359) to that other ticket.

**Comment by rjohnson2011** [comment]
@GnatalieH Just checking in to see if there's any updates regarding the design of these buttons.

**Comment by bkjohnson** [comment]
It seems reasonable to me. I can't think of any drawbacks at the moment.

**Comment by GnatalieH** [comment]
@rjohnson2011 I just want to make sure this is cool with @noahgelman. Noah, is it acceptable for all of our buttons to use the darker hover color in addition to the focus halo when a button is focused on?

**Comment by noahgelman** [comment]
@GnatalieH @bkjohnson A darker hover color is fine on focus. No accessibility issue.

**Comment by GnatalieH** [comment]
@bkjohnson what do we need to do to get this pattern in the design system?

**Comment by bkjohnson** [comment]
It looks like we already have a color that we could use for the Default button: `#003e73 ` ([`primary-darker`](https://design.va.gov/design/color-palette))

_(I think I got that right - it's a bit awkward that our Primary button uses the green color, and the Default button uses the primary color)_

---

I took a closer look and the buttons in the screenshot are using styles from `uswds`.

![image](https://user-images.githubusercontent.com/2008881/134262842-57f425a1-8326-4f9d-a321-31c59e9a4ec7.png)

However, notice how the button that _does_ have the appropriate `:focus` color change is using the `va-button-primary` class

![image](https://user-images.githubusercontent.com/2008881/134263127-612d7507-a209-48ab-b611-8cdaf3eb1a73.png)

Our options include:

1. Override uswds button styles [in this file](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/blob/master/packages/formation/sass/modules/_m-button.scss)
1. Add more styles for `va-button-default:hover` alongside [these classes](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/blob/93576939aa1b57950fd2c30b76cf3ca2cf66d383/packages/formation/sass/base/_va.scss#L1046-L1071) and ensure that they get used
1. Create a web component that can be used like `<va-button primary>` or `<va-button secondary>` with all the appropriate styles automatically applied.

I'm in favor of option 3 because having to override CSS from a global dependency is annoying, and option 2 would be a poor experience for VFS developers. Options 1 and 2 are also more difficult to maintain in my opinion. `_shame.scss` is a testament to the maintenance difficulty.

**Comment by GnatalieH** [comment]
@bkjohnson can we change those names so they accurately describe the button/color if we go with option 3? I think this proposed change in general is not a significant upgrade, so if we decide it's too much effort to fix and maintain, I'm OK with that too.

**Comment by bkjohnson** [comment]
We could, that might be more @rtwell 's call though.

**Comment by pjhill** [comment]
> We could, that might be more @rtwell 's call though.

@rtwell would you mind weighing in on the options presented in Brooks' [previous comment](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/235#issuecomment-924478996)?

**Comment by GnatalieH** [comment]
Hi. I met with @rtwell and this is his feedback:

- Yes, we should move forward with using the hover color (plus focus halo) on the focus state for all of the button types. It's an affordance.
- Focus halo color: please make sure the focus halo we use is the new color identified by our A11y experts (@artsymartha68, @noahgelman). 
- In terms of naming and CSS conventions, the Developers can decide how to handle things in the code. I am going to re-assess how buttons are named in our documentation at a later date.

CC: @pjhill, @bkjohnson

**Comment by rjohnson2011** [comment]
I'm going to change the CSS since that's a quick fix that takes care of the hover and focus. 
Next I'll create a new ticket to address creating web components based on all your feedback.

**Comment by rjohnson2011** [comment]
![image](https://user-images.githubusercontent.com/72466113/135493026-31b671df-f9a1-433c-b746-a1eae3db42e8.png)
Focus & Hover now have same background for Secondary Buttons.
![image](https://user-images.githubusercontent.com/72466113/135493291-cef8ac75-627c-423e-b3ae-c1027643f50d.png)
Blue Default buttons now have same background color on hover and focus.
